tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,288 @@
+        """
+        Sets Constants for all the Tests, fixtures which will be later used for assertions
+        """
+        request.cls.region = settings.azurerm.azure_region
+        request.cls.rhel7_ft_img = AZURERM_RHEL7_FT_IMG_URN
+        request.cls.rg_default = AZURERM_RG_DEFAULT
+        request.cls.premium_os_disk = AZURERM_PREMIUM_OS_Disk
+        request.cls.platform = AZURERM_PLATFORM_DEFAULT
+        request.cls.vm_size = AZURERM_VM_SIZE_DEFAULT
+        request.cls.hostname = gen_string('alpha')
+        request.cls.fullhostname = '{}.{}'.format(self.hostname, module_domain.name).lower()
+
+        request.cls.compute_attrs = {
+            "resource_group": self.rg_default,
+            "vm_size": self.vm_size,
+            "username": settings.azurerm.username,
+            "password": settings.azurerm.password,
+            "platform": self.platform,
+            "script_command": 'touch /var/tmp/text.txt',
+            "script_uris": AZURERM_FILE_URI,
+            "image_id": self.rhel7_ft_img,
+        }
+
+        nw_id = module_azurerm_cr.available_networks()['results'][-1]['id']
+        request.cls.interfaces_attributes = {
+            "0": {
+                "compute_attributes": {
+                    "public_ip": "Dynamic",
+                    "private_ip": "false",
+                    "network": nw_id,
+                }}}
+
+    @pytest.fixture(scope='class')
+    def class_host_ft(self,
+                      azurermclient,
+                      module_azurerm_finishimg,
+                      module_azurerm_cr,
+                      module_architecture,
+                      module_domain,
+                      module_location,
+                      module_org,
+                      module_os,
+                      module_smart_proxy,
+                      module_puppet_environment):
+        """
+        Provisions the host on AzureRM using Finish template
+        Later in tests this host will be used to perform assertions
+        """
+
+        skip_yum_update_during_provisioning(template='Kickstart default finish')
+        host = entities.Host(
+            architecture=module_architecture,
+            compute_resource=module_azurerm_cr,
+            compute_attributes=self.compute_attrs,
+            interfaces_attributes=self.interfaces_attributes,
+            domain=module_domain,
+            organization=module_org,
+            operatingsystem=module_os,
+            location=module_location,
+            name=self.hostname,
+            provision_method='image',
+            image=module_azurerm_finishimg,
+            root_pass=gen_string('alphanumeric'),
+            environment=module_puppet_environment,
+            puppet_proxy=module_smart_proxy,
+            puppet_ca_proxy=module_smart_proxy,
+        ).create()
+        yield host
+        skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
```

Comment:
Why is host provisioned in fixture teardown? Shouldn't that be done before `yield host`? Or just skipped altogether?

Reply:
after the yield statement serves as the teardown code :)

Reply:
I know, I asked for a reason why.
But I know now, we first break kickstart template, and then feel compelled to return it back to what Robottelo thinks it was. There are obvious issues with that approach, starting with race conditions, but I see you just follow pattern established elsewhere.

Reply:
ahh ..okay ..  so could you please let me know in which case we could see race condition?

Reply:
Let's say we have two tests following that pattern running in parallel. Following sequence of events is possible:
1. Test A does `skip_yum_update_during_provisioning(template='Kickstart default finish')`
2. Test B does `skip_yum_update_during_provisioning(template='Kickstart default finish')` (effectively no-op, I presume)
3. Test A does it's thing
4. Test B does expensive operation
5. Test A finishes and runs teardown, doing `skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)`
6. Test B does it's thing, still thinking kickstart modification is in effect; it's unaware that Test A reverted that in the meantime.

Obviously, each test should have it's own kickstart template with content it needs. Right now we have bunch of random tests modifying global state at will.

As I said, I see you just reused pattern established elsewhere, so I don't expect you to fix that as part of your PR. But if you wan't to, be my guest.

Reply:
Agree with you on it ...but first would like to observe the automation results around these tests and then would put a call to clone the existing template and run `skip_yum_update_during_provisioning` on that .